src/LocalNeuralNet.cpp

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2015 - 2022, Intel Corporation
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include "config.h"
+#include "LocalNeuralNet.hpp"
+
+#include <cmath>
+
+#include "geopm/Exception.hpp"
+#include "TensorOneD.hpp"
+#include "TensorTwoD.hpp"
+
+
+namespace geopm
+{
+    LocalNeuralNet::LocalNeuralNet(const LocalNeuralNet &other)
+        : m_layers(other.m_layers)
+    {
+    }
+
+    LocalNeuralNet::LocalNeuralNet(json11::Json input)
+    {
+        if (!input.is_array()) {
+            throw geopm::Exception("Neural network weights is non-array type.\n",
+                                   GEOPM_ERROR_INVALID, __FILE__, __LINE__);
+        }
+        if (input.array_items().size() == 0u) {
+            throw geopm::Exception("Empty array is invalid for neural network weights.\n",
+                                   GEOPM_ERROR_INVALID, __FILE__, __LINE__);
+        }
+
+        m_layers.resize(input.array_items().size());
+        for (std::size_t idx = 0; idx < m_layers.size(); idx++) {
+            if (!input[idx].is_array() || input[idx].array_items().size() != 2) {
+                throw geopm::Exception("Neural network weight entries should be doubles.\n",
+                                       GEOPM_ERROR_INVALID, __FILE__, __LINE__);
+            }
+
+            m_layers[idx] = std::make_pair(TensorTwoD(input[idx][0]), TensorOneD(input[idx][1]));
+            if (m_layers[idx].first.get_rows() != m_layers[idx].second.get_dim()) {
+                throw geopm::Exception("Incompatible dimensions for weights and biases.",
+                                       GEOPM_ERROR_INVALID, __FILE__, __LINE__);
+            }
+            if (idx > 0 && m_layers[idx].first.get_cols() != m_layers[idx-1].first.get_rows()) {
+                throw geopm::Exception("Incompatible dimensions for consecutive layers.",
+                                       GEOPM_ERROR_INVALID, __FILE__, __LINE__);
+            }
+        }
+    }
+
+    LocalNeuralNet&
+    LocalNeuralNet::operator=(const LocalNeuralNet &other)
+    {
+        m_layers = other.m_layers;
+
+        return *this;
+    }
+
+    TensorOneD
+    LocalNeuralNet::model(TensorOneD inp)
```

Comment:
Minor nitpick, but the return type on the line above looks maybe unconventional code style. More importantly, however pass the class as const reference to avoid an extra copy.
```cpp
TensorOneD LocalNeuralNet::model(const TensorOneD &inp)
```